### ISSUE TITLE
Missing translation strings (CashFusion)

### ISSUE THREAD

**Issue by georgengelmann** [Electron-Cash/Electron-Cash#1804]
Some CashFusion messages are not on Crowdin [it's because they come from low-level error messages - Calin]

Examples:
"No outputs available at any Tier."
"failed"
"waiting"

![photo_2020-02-09_13-07-58](https://user-images.githubusercontent.com/13990731/74176060-dbea6680-4c2e-11ea-9d19-cd7029ff6129.jpg)

**Comment by cculianu** [comment]
@markblundeberg once we are happy with the various strings we can set up a static string dict similar to pr_tooltips which lives in gui/qt/útil.py with “fake” _() decorators so gettext picks them up.  Then we have to translate again at runtime (because static strings are sometimes evaluated before i18n is set up).

Hit me up and I can even do that for you.

Thanks @georgengelmann for reporting so we don’t forget.

**Comment by markblundeberg** [comment]
The status field is an internal logic thing so it will have to be dynamically translated at display (there are only a few codes so it's no biggie), however the extra status thing is purely for human reading so we can translate at generation time.

@cculianu my understanding is that `N_("deferred translation")` is the canonical way to do that kind of thing... https://docs.python.org/3/library/gettext.html#deferred-translations ... perhaps that one part of gui/qt/util.py should be updated to use N_ instead of _.

**Comment by cculianu** [comment]
@markblundeberg Ah, I didn't know about the `N_()` thing.  So if gettext knows about that, great.  Yeah we can do it that way.  When I'm done with what I'm working on now, I'll go in and edit all the deferred translations to use that scheme, and then we can do similar for the GUI strings.  Convenient, since now you can do `N_()` in-place and then before display we run them through `_()` again in `qt.py`. 👍

**Comment by georgengelmann** [comment]
Not in CashFusion - e-mail window (Cancel / OK button)

![Screenshot at 2020-02-13 13:24:07](https://user-images.githubusercontent.com/13990731/74435318-1422bc80-4e5c-11ea-9059-b14931c7fd8c.png)

**Comment by cculianu** [comment]
The latter needs the string `&OK` and `C&ancel` translated... (they are in crowin)

**Comment by EchterAgo** [comment]
The problem with the "Ok" and "Cancel" text is that they come from Qt and last time I checked we can not load Qt translations due to a bug in PyQt5 where they don't distribute all necessary files.

**Comment by cculianu** [comment]
Interesting -- because for that bit of code it looks like they load custom QPushButtons -- you're saying if Qt sees `&OK` or whatever it generates some generic button?

**Comment by EchterAgo** [comment]
No, this is a standard `QInputDialog.getText` in `plugins/email_requests/qt.py:205`.

https://doc.qt.io/qt-5/qinputdialog.html
https://doc.qt.io/qt-5/images/inputdialogs.png

**Comment by cculianu** [comment]
Oh yeah you're right.. I missed that.  Later on in the file they use the custom class we have called `Buttons` so I thought it was that.

Yeah ok so the fix to this is just use the custom classes we have over Qt's... Noted.

**Comment by EchterAgo** [comment]
The `Buttons` class sadly does not handle platform specific standard button layouts / names like `QDialogButtonBox` does:

https://doc.qt.io/qt-5/qdialogbuttonbox.html

We probably should replace our `Buttons` implementation with one that uses `QDialogButtonBox` as a base and getting Qt translation working. It should be possible, though we might have to copy some files around. I'll look into this soon.

**Comment by cculianu** [comment]
I'm skeptical.  

Advantage:
  -  We get to use Qt's amazing QDialogButtonBox class. Which for us just amounts to OK/Cancel being on left or right swapped depending on if Windows or if !Windows.

 Disadvantage: 
  - We add some more complexity to our packaging system

Another approach is just to hack Buttons() class to do this and not have to add more complexity to our packaging to deal with Qt translations.

However if you can figure out a way to do it "cheaply" (that is, just leveraging what we already have in terms of gettext stuff), ok. 

I'm skeptical though. :)
